pkg/plugins/driverutil/config.go

```diff
@@ -311,6 +318,18 @@ func (ct *DefaultingConfigTracker[T]) History(ctx context.Context, target Target
 	return revisions, nil
 }
 
+func (ct *DefaultingConfigTracker[T]) runValidation(conf T) (*protovalidate.ValidationError, error) {
+	err := ct.validator.Validate(conf)
+	var valErr *protovalidate.ValidationError
+	if err != nil {
+		if !errors.As(err, &valErr) {
+			// invalid validation rules, etc.
+			return nil, status.Error(codes.Internal, err.Error())
```

Comment:
In what cases would we encounter invalid validation rules? Does protobuf generation pass if we have invalid generation rules?

Reply:
Currently, invalid cel expressions are not a compiler error, but the upstream typed builtin options are. They are validated, I just haven't implemented surfacing the warnings to the diagnostic handler yet so that they can be handled like actual compiler errors. 
For now, invalid expressions won't have the embedded syntax highlighting, like this:
![image](https://github.com/rancher/opni/assets/8194899/378d3bc1-733c-4d35-b37f-c58c0e0c12f0)
![image](https://github.com/rancher/opni/assets/8194899/1386b968-200c-4c1f-b516-f9cf3ce969ca)

and the cel errors are logged in the `Protobuf Language Server` vscode output:
```
ERROR: <input>:1:50: Syntax error: token recognition error at: '= '
 | (this.type == 0 && has(this.etcd)) || (this.type = 1 && has(this.jetstream))
 | .................................................^
```